include/sortcheck.h

```diff
@@ -30,9 +30,20 @@ namespace sortcheck {
 #endif
 
 // TODO: replace with std:less ?
-template<typename T>
-struct Compare {
-  bool operator()(const T &lhs, const T &rhs) const {
+template<typename A>
+struct Compare1 {
+  bool operator()(const A &lhs, const A &rhs) const {
```

Comment:
It is not really needed but probably can improve some optimizations
```suggestion
  bool operator()(const A &lhs, const A &rhs) const noexcept(noexcept(lhs < rhs)) {
```
WDYT?

Reply:
Good idea but will need to wrap it in macro, for C++98 compatibility.